src/python/foglamp/core/scheduler.py

```diff
@@ -547,60 +791,431 @@ def _schedule_next_task(self, schedule):
+                             self._ScheduleType.TIMED] and (
+                is_new_schedule or
+                prev_schedule_row.time != schedule_row.time or
+                prev_schedule_row.day != schedule_row.day or
+                prev_schedule_row.repeat_seconds != schedule_row.repeat_seconds or
+                prev_schedule_row.exclusive != schedule_row.exclusive):
+            self._schedule_first_task(schedule_row, time.time())
+            self._resume_check_schedules()
+
+    async def delete_schedule(self, schedule_id: uuid.UUID):
+        """Deletes a schedule
+
+        Args:
+            schedule_id
+
+        Raises:
+            ScheduleNotFoundError
+
+            NotReadyError
+        """
+        if not self._ready:
+            raise NotReadyError()
+
+        # TODO: Inspect race conditions with _set_first
+        try:
+            del self._schedules[schedule_id]
+        except KeyError:
+            raise ScheduleNotFoundError(schedule_id)
+
+        async with aiopg.sa.create_engine(_CONNECTION_STRING) as engine:
+            async with engine.acquire() as conn:
+                await conn.execute(self._schedules_tbl.delete().where(
+                    self._schedules_tbl.c.id == str(schedule_id)))
+
+    @classmethod
+    def _schedule_row_to_schedule(cls,
+                                  schedule_id: uuid.UUID,
+                                  schedule_row: _ScheduleRow) -> Schedule:
+        schedule_type = schedule_row.type
+
+        if schedule_type == cls._ScheduleType.STARTUP:
+            schedule = StartUpSchedule()
+        elif schedule_type == cls._ScheduleType.TIMED:
+            schedule = TimedSchedule()
+        elif schedule_type == cls._ScheduleType.INTERVAL:
+            schedule = IntervalSchedule()
+        elif schedule_type == cls._ScheduleType.MANUAL:
+            schedule = ManualSchedule()
+        else:
+            raise ValueError("Unknown schedule type {}", schedule_type)
 
-            self._main_sleep_task = None
+        schedule.schedule_id = schedule_id
+        schedule.exclusive = schedule_row.exclusive
+        schedule.name = schedule_row.name
+        schedule.process_name = schedule_row.process_name
+        schedule.repeat = schedule_row.repeat
 
-    def start(self):
+        if schedule_type == cls._ScheduleType.TIMED:
+            schedule.day = schedule_row.day
+            schedule.time = schedule_row.time
+
+        return schedule
+
+    async def get_scheduled_processes(self) -> List[ScheduledProcess]:
+        """Retrieves all rows from the scheduled_processes table
+        """
+        if not self._ready:
+            raise NotReadyError()
+
+        processes = []
+
+        for (name, script) in self._process_scripts.items():
+            process = ScheduledProcess()
+            process.name = name
+            process.script = script
+            processes.append(process)
+
+        return processes
```

Comment:
There won't be a straight-forward conversion between the objects in the schedule module to JSON. Secondly, having an ordered list lets me write better tests. In fact because of this I found bugs that I wouldn't have found otherwise.